api/src/paths/project/{projectId}/survey/{surveyId}/observation/submission/upload.ts

```diff
@@ -191,6 +199,61 @@ export const insertSurveyOccurrenceSubmission = async (
   return insertResponse;
 };
 
+/**
+ * Inserts a new record into the `occurrence_submission` table.
+ *
+ * @param {number} surveyId
+ * @param {string} source
+ * @param {string} key
+ * @param {IDBConnection} connection
+ * @return {*}  {Promise<void>}
+ */
+export const getTemplateMethodologySpeciesId = async (
+  surveyId: number,
+  file_name: string,
+  connection: IDBConnection
+): Promise<number | null> => {
+  let templateName;
+
+  switch (file_name) {
+    case 'Moose_SRB_or_Composition_Survey_Skeena.xlsx':
+      templateName = 'Moose SRB or Composition Survey Skeena';
+      break;
+    case 'Moose_SRB_or_Composition_Survey_Omineca.xlsx':
+      templateName = 'Moose SRB or Composition Survey Skeena';
+      break;
+    case 'Moose_SRB_or_Composition_Survey_Cariboo.xlsx':
+      templateName = 'Moose SRB or Composition Survey Skeena';
+      break;
+    case 'Moose_SRB_or_Composition_Survey_Okanagan.xlsx':
+      templateName = 'Moose SRB or Composition Survey Skeena';
+      break;
+    case 'Moose_SRB_or_Composition_Survey_Kootenay.xlsx':
+      templateName = 'Moose SRB or Composition Survey Kootenay';
+      break;
+    case 'Moose_Recruitment_Survey.xlsx':
+      templateName = 'Moose Recruitment Survey';
+      break;
+  }
+
+  if (templateName) {
+    const getIdSqlStatement = getTemplateMethodologySpeciesIdSQL(surveyId, templateName);
+
+    if (!getIdSqlStatement) {
+      throw new HTTP400('Failed to build SQL get Id statement');
+    }
+    const getIdResponse = await connection.query(getIdSqlStatement.text, getIdSqlStatement.values);
+
+    if (!getIdResponse) {
+      throw new HTTP400('Failed to query template methodology species table');
+    }
+
+    return getIdResponse?.rows?.[0]?.template_methodology_species_id || null;
```

Comment:
question: do we even need the `|| null` in here since you already have all the `?` in place so wouldn't it just return null if property doesnt exist or the actual value?

Reply:
I believe it will return `undefined` if it didnt exist, rather than null.  We can obv handle both values, but its a bit nicer to intentionally return `null` I think

Reply:
gotcha, wasn't sure if the return value was undefined/null or if it mattered what we returned - if we want to return null then this is fine 👍🏼